float32 freq 
bool active
---
bool ack
